authorization/src/lib.rs

```diff
@@ -5,39 +5,37 @@ extern crate std;
 
 mod test;
 
-use soroban_sdk::{contractimpl, contracttype, BigInt, Env, IntoVal, Symbol};
+use soroban_sdk::{contractimpl, contracttype, symbol, BigInt, Env, IntoVal};
 use soroban_sdk_auth::{
     check_auth, NonceAuth, {Identifier, Signature},
 };
 
-#[derive(Clone)]
 #[contracttype]
 pub enum DataKey {
-    Acc(Identifier),
+    SavedNum(Identifier),
     Nonce(Identifier),
     Admin,
 }
 
 fn read_nonce(e: &Env, id: Identifier) -> BigInt {
     let key = DataKey::Nonce(id);
-    if let Some(nonce) = e.contract_data().get(key) {
-        nonce.unwrap()
-    } else {
-        BigInt::zero(e)
-    }
+    e.contract_data()
+        .get(key)
+        .unwrap_or_else(|| Ok(BigInt::zero(e)))
+        .unwrap()
 }
-struct WrappedAuth(Signature);
 
-impl NonceAuth for WrappedAuth {
+struct NonceForSignature(Signature);
+
+impl NonceAuth for NonceForSignature {
```

Comment:
`WrappedAuth` looks like something that provides access to the nonce for a `Signature`. So I suggest we name it `NonceForSignature` instead. WrappedX tells us information about the code structure, rather than telling us what the concept and intent is.